src/form_urlencoded.rs

```diff
@@ -276,21 +276,29 @@ impl<'a, T: Target> Serializer<'a, T> {
     pub fn extend_pairs<I, K, V>(&mut self, iter: I) -> &mut Self
     where
         I: IntoIterator,
-        I::Item: Borrow<(K, V)>,
+        I::Item: Borrow<(K, Option<V>)>,
```

Comment:
NB: Despite my potentially-misleading previous commit message, this interface is *not* currently backwards-compatible.  It feels like it should be possible to maintain a compatible API, but I'm not yet certain about the best order of steps to achieve that from here.

Observations include:

- That the underlying [name, value iterator](https://github.com/servo/rust-url/blob/28314752bf138a87bd8151f610d385629dec5710/src/form_urlencoded.rs#L46-L55) could be refactor to return a stream of tokens (see a [similar proposed refactor in cpython](https://github.com/python/cpython/pull/19949/files#diff-01126e4d609357e876a7fd7baa62ce18R748-R779))

- That it may be possible to define [shared behaviour](https://doc.rust-lang.org/book/ch10-02-traits.html#traits-defining-shared-behavior) on a common base parser trait, and then allow call-sites to explicitly opt-in to override and enable parsing of standalone keys